### ISSUE TITLE
Errors with SQL

### ISSUE THREAD

**Issue by KeiroD** [OpenImporter/openimporter#58]
Hi,

Sorry, the database connection information used in the specified installation of your forum cannot access the installation of SMF 2.0. This may either mean that the installation doesn't exist, or that the MySQL account used does not have permissions to access it.

The error MySQL gave was: SELECT command denied to user 'forums'@'localhost' for table 'smf_members''
error_line 329
error_file /home/user/public_html/domain.com/derp/OpenImporter/Importer.php

Note: Your OpenImporter setup doesn't allow for specifying db name and user. This needs to be changed, as it basically assumes by default that what Elkarte uses is what SMF uses in terms of db username/password.

**Comment by margarett** [comment]
That's how SMF converters work and it's a "heritage" that OpenImporter also has...
It uses the destination's database user info to access both databases...

Having performed dozens of conversions over the last few years, I seriously think that this is a non-issue... Any decent host (heck, even most crappy ones! :tongue: ) has the capability to add several users to the same database. I don't know how much work it represents (to have 2 authentications) but if it's more than 10 minutes to implement, it's probably not worth it...

**Comment by emanuele45** [comment]
Yup, true: the user/pwd pair used is the one of the "destination" install.
And it's true, change that behaviour to use the correct pair would be great.

I though I opened an issue previously, but probably I just mentioned in some topic... or just plain forgot. xD
Anyway, it will probably be implemented easily after moving to a Doctrine or PDO as per #47

**Comment by KeiroD** [comment]
"That's how SMF converters work and it's a "heritage" that OpenImporter also has...
It uses the destination's database user info to access both databases..."

I was unaware of that, actually... this is my first time doing an import. Good to know, though... I did notice it was trying to, or at least it appeared to, automatically use both Settings.php to access both forums and transfer things over by itself automatically.

But assuming the user/pwd pair used of the destination install for accessing the origin db of the forum is... kinda not exactly good practice. Especially if it wasn't immediately obvious that you should grant access to the destination sql user to the origin forum for purposes of access and transfer.

**Comment by KeiroD** [comment]
Sidenote: Now that I know that the importer requires that the destination forum user be added to the old forum for access, it works perfectly.

If you guys haven't included this data on the Importer, you really, really should.

**Comment by margarett** [comment]
@emanuele45, should the 2 database users stuff be working already? I would say no but I'm using root with blank password @localhost so... :tongue:

**Comment by emanuele45** [comment]
At the "DBAL" level it **should**: OI tries to establish two different connections with the data from the source and the destination. But honestly I still have to test it properly.
There are a couple of "indirect" problems due to #60.
